clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1747,11 +1751,17 @@ private String updatePushJobDetailsWithJobDetails(DataWriterTaskTracker dataWrit
     final long recordTooLargeFailureCount = dataWriterTaskTracker.getRecordTooLargeFailureCount();
     if (recordTooLargeFailureCount > 0) {
       updatePushJobDetailsWithCheckpoint(PushJobCheckpoints.RECORD_TOO_LARGE_FAILED);
-      String errorMessage = String.format(
-          "Input data has at least %d records that exceed the maximum record limit of %s",
+
+      // maxSizeForUserPayloadPerMessageInBytes < 1MB (Kafka record limit); errors appear when chunking is not enabled
+      // maxRecordSizeBytes can be much larger and should take effect only once chunking is enabled
+      final int recordSizeLimit = (pushJobDetails.chunkingEnabled)
+          ? getVeniceWriter(pushJobSetting).getMaxRecordSizeBytes()
+          : getVeniceWriter(pushJobSetting).getMaxSizeForUserPayloadPerMessageInBytes();
+      final String errorMessage = String.format(
+          "Input data has at least %d records that exceed the maximum %s record limit of %s",
           recordTooLargeFailureCount,
-          generateHumanReadableByteCountString(
-              getVeniceWriter(pushJobSetting).getMaxSizeForUserPayloadPerMessageInBytes()));
+          (pushJobDetails.chunkingEnabled) ? "Venice" : "Kafka",
```

Comment:
Let's not confuse the users with different error messages. We don't really need the system name here

Reply:
Removed.

I actually wanted this mainly for our debugging purposes to determine whether the failure is due to the `~1MB` limit or the new `maxRecordSizeBytes` limit.

I find it a lot clearer when the system name is mentioned, otherwise we would have to look at the record limit number in the error message to determine which version of the failure is being hit.